addons/sale/static/src/js/selection_kanban/selection_kanban_field.js

```diff
@@ -0,0 +1,57 @@
+/** @odoo-module **/
+
+import { _t } from "@web/core/l10n/translation";
+import { registry } from "@web/core/registry";
+import { SelectionField, selectionField } from "@web/views/fields/selection/selection_field";
+
+export class SelectionKanbanField extends SelectionField {
+    static template = "sale.SelectionKanbanField";
+    static props = {
+        ...SelectionField.props,
+        autosave: { type: Boolean, optional: true },
+    };
+
+    /**
+     * @param {Event} ev
+     */
+    onChange(ev) {
+        const value = JSON.parse(ev.target.value);
+        switch (this.type) {
+            case "many2one":
+                if (value === false) {
+                    this.props.record.update({ [this.props.name]: false }, { save: this.props.autosave });
+                } else {
+                    this.props.record.update({
+                        [this.props.name]: this.options.find((option) => option[0] === value),
+                    }, { save: this.props.autosave });
+                }
+                break;
```

Comment:
If the use is restricted to sale, should we really keep the support of m2o fields ?
Shouldn't we simply update the supportedTypes to restrict to selection fields, which is our only usecase for now?

Reply:
Well, we don't have to. I restricted it to `selection` as you suggested